imagepipeline/src/main/java/com/facebook/imagepipeline/producers/ResizeAndRotateProducer.java

```diff
@@ -237,19 +256,21 @@ private static TriState shouldTransform(
     final float heightRatio = ((float) resizeOptions.height) / height;
     float ratio = Math.max(widthRatio, heightRatio);
 
-    // TODO: The limit is larger than this on newer devices. The problem is to get the real limit,
-    // you have to call Canvas.getMaximumBitmapWidth/Height on a real HW-accelerated Canvas.
-    if (width * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / width;
-    }
-    if (height * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / height;
-    }
+    // TODO: Both width and height should be equal or smaller than device's max bitmap size
```

Comment:
This is not the right place to do this clamping as this returns float value that later gets round, and can therefore get round up.